### PR TITLE
[test] De-gyb stdlib unittest.

### PR DESCRIPTION
**Pull request by milseman**
StdlibUnittest uses gyb to avoid duplicating many source-context
arguments. However, this means that any test that wishes to add new
expect helpers has to also be gybbed. Given that this structure hasn't
changed in years, and we should have a real language support
eventually, de-gyb it.

Otherwise tests have to choose between gybbing themselves, not writing
expectation helpers, or hard-code traces and risk divergence over
time. De-gybbing is the least of these evils.

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please test

**Review by lorentey — APPROVED** [review]
👍This is great! I'll now be able to use these without spending half an hour on relearning the gyb magic every. single. time.

LGTM, except: could you please also reindent the degybbed files? Some of the indentation is messed up.

Note that I'm planning to land some changes to some of these files soon in #15939; if I get that done today, I may generate some conflicts. :(  (Don't wait for that PR, though -- I'd gladly redo my changes on the degybbed files.)

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionInstance.swift [hunk 1]
```diff
@@ -0,0 +1,1917 @@
...
+  Instances.Element == BaseCollection.Index {
+
+  checkEquatable(instances, oracle: equalityOracle, message(),
+  stackTrace: stackTrace.pushIf(showFrame, file: file, line: line))
+  for i in instances {
+    if i != startIndex {
+      let next = baseCollection.index(before: i)
+      // index(before:) gets us a new index value
+      expectNotEqual(i, next, message(),
+  stackTrace: stackTrace.pushIf(showFrame, file: file, line: line))
```
[the remark this answers is not in the document]
Comment (milseman) [reply stdlib/private/StdlibCollectionUnittest/CheckCollectionInstance.swift]: Ugh, I fixed many by hand but probably missed a few

[the remark this answers is not in the document]
Comment (milseman) [reply stdlib/private/StdlibCollectionUnittest/CheckCollectionInstance.swift]: (we really need a swift-format tool we just blindly apply to all code)

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
Whoops, deleted a label when indenting.

**Comment by milseman** [comment]
@swift-ci please test

**Pull request merged**